x/whitelist/keeper/keeper.go

```diff
@@ -0,0 +1,111 @@
+package keeper
+
+import (
+	"bytes"
+
+	"github.com/cosmos/cosmos-sdk/codec"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	gogotypes "github.com/gogo/protobuf/types"
+
+	"github.com/Sifchain/sifnode/x/whitelist/types"
+)
+
+type keeper struct {
+	cdc codec.BinaryMarshaler
+
+	storeKey sdk.StoreKey
+	// bankKeeper    types.BankKeeper
+}
+
+func NewKeeper(cdc codec.Marshaler, storeKey sdk.StoreKey) types.Keeper {
+	return keeper{
+		// bankKeeper: bankKeeper,
+		cdc:      cdc,
+		storeKey: storeKey,
+	}
+}
+
+func (k keeper) SetAdminAccount(ctx sdk.Context, adminAccount sdk.AccAddress) {
+	store := ctx.KVStore(k.storeKey)
+	key := types.AdminAccountStorePrefix
+	store.Set(key, k.cdc.MustMarshalBinaryBare(&gogotypes.BytesValue{Value: adminAccount}))
+}
+
+func (k keeper) IsAdminAccount(ctx sdk.Context, adminAccount sdk.AccAddress) bool {
+	account := k.GetAdminAccount(ctx)
+	if account == nil {
+		return false
+	}
+	return bytes.Equal(account, adminAccount)
+}
+
+func (k keeper) GetAdminAccount(ctx sdk.Context) (adminAccount sdk.AccAddress) {
+	store := ctx.KVStore(k.storeKey)
+	key := types.AdminAccountStorePrefix
+	bz := store.Get(key)
+	acc := gogotypes.BytesValue{}
+	k.cdc.MustUnmarshalBinaryBare(bz, &acc)
+
+	adminAccount = sdk.AccAddress(acc.Value)
+
+	return adminAccount
+}
+
+func (k keeper) GetDenom(ctx sdk.Context, denom string) types.DenomWhitelistEntry {
+	wl := k.GetDenomWhitelist(ctx)
+
+	for i := range wl.DenomWhitelistEntries {
+		if wl.DenomWhitelistEntries[i].Denom == denom &&
+			wl.DenomWhitelistEntries[i] != nil {
+			return *wl.DenomWhitelistEntries[i]
+		}
+	}
+
+	return types.DenomWhitelistEntry{
```

Comment:
Cant we just return bool ?

Reply:
Oh is it becuase you want to use Units in the future ?

Reply:
It's so that we can include a decimal value in the denom in future and use from clp.